scripts/m18_submission_checker/aida_m18_submission_checker.py

```diff
@@ -0,0 +1,383 @@
+            
+        '''
+        #type count and ta1_temp used to determine if 1a or 1b
+        type_count = 0
+        ta1_ret_val = 0
+
+        names_str = " ".join(ARCHIVE_NAMES)
+        ta1a_dir_struct_regex = re.compile(".*\/NIST\/[^\/]+.ttl", re.IGNORECASE)
+        ta1b_dir_struct_regex = re.compile(".*\/NIST\/\S*\/\S*\.ttl", re.IGNORECASE)
+
+        #if ttls located in the NIST directory, mark as potential TA_1a
+        if ta1a_dir_struct_regex.search(names_str) is not None:
+            ta1_ret_val = 0
+            type_count = type_count + 1
+        #if ttls located at some subdirectory of the NIST directory, mark as potential TA_1b
+        if ta1b_dir_struct_regex.search(names_str) is not None:
+            ta1_ret_val = 1
+            type_count = type_count + 1
+        #if ttls were found in both the NIST/ directory and a subdirectory of NIST/, mark as TA_1a and have the ta_1a do this check too
+        if ta1_ret_val > -1 and type_count > 1 :        #use > and not == for scaleability? idk dont @ me c:
+            logging.warning("WARNING: Subdirectories found at the same level of .ttl files in the NIST directory.")
+            ta1_ret_val = 0
+        return ta1_ret_val
+    elif dot_count > 1 and dot_count < 4:
+        return dot_count
+    else:
+        logging.error("Cannot figure out the TA level. Please ensure archive name is properly titled per NIST guidelines.")
+        return -1
+
+
+
+def validate_archive_filetype():
+    """
+    Checks the archive file type. If it is not a valid filetype, the program will
+    print an error message and quit the script.
+    Valid filetypes: .tgz, .tar.gz, or .zip.
+
+    :return: True if valid filetype and successfully set the global FILETYPE variable, False otherwise
+    :rtype: bool
+    """
+    global FILETYPE
+
+    if ARCHIVE_FILE_NAME.endswith('.tar.gz') or ARCHIVE_FILE_NAME.endswith('.tgz'):
+        FILETYPE = 0
+    elif ARCHIVE_FILE_NAME.endswith('.zip'):
+        FILETYPE = 1
+    else:
+        FILETYPE = -1
+        return False
+
+    return True
+
+def main():
+    """
+
+    :returns: True if script completed successfully, False otherwise
+    :rtype: bool
+    """
+    global FILETYPE
+    global TASK_TYPES
+    global ARCHIVE_NAMES
+    global ARCHIVE_FILE_NAME
+
+    # set logging to log to stdout
+    logging.basicConfig(level='INFO', format='%(levelname)s:  %(message)s')
+
+    #check runtime arguments
+    if len(sys.argv) < 2:
+        logging.error("Missing archive file as argument.\nTo run: python3 aida_m18_submission_checker.py <path/to/archive>")
+        return False
+    if not os.path.isfile(sys.argv[1]):
+        logging.error("Argument is not a file.\nTo run: python3 aida_m18_submission_checker.py <path/to/archive>")
+        return False
+
+    ARCHIVE_FILE_NAME = str(sys.argv[1])
+
+    #ensure the archive type is a valid filetype and set the global FILETYPE value
+    if validate_archive_filetype() < 0:
```

Comment:
This is a bug, the `validate_archive_filetype()` returns a boolean, not an int. Update this to be 
`if not validate_archive_filetype():`